spinnman/model/executable_targets.py

```diff
@@ -32,21 +49,39 @@ def add_subsets(self, binary, subsets):
                 self._total_processors += 1
                 self._all_core_subsets.add_processor(subset.x, subset.y, p)
 
-    def add_processor(self, binary, chip_x, chip_y, chip_p):
+        # add to binary to executable type
+        self._binary_to_executable_types_map[executable_type].add(binary)
+
+    def add_processor(self, binary, chip_x, chip_y, chip_p, executable_type):
```

Comment:
Add a default for `executable_type` (to the most common form in our users' code?) to reduce the possibility of incompatibilities.

Reply:
as above